### PR TITLE
Disabled auto load libs

### PR DESCRIPTION
**Pull request by mohitrpatil**
please review and approve.

### CODE REVIEW

File: tests/test_cdg.py [hunk 1]
```diff
@@ -17,9 +17,7 @@ def test_graph_0():
     # etc.
 
     # Create a project with a random binary - it will not be used anyways
-    p = angr.Project(os.path.join(test_location, 'x86_64', 'datadep_test'),
-                     load_options={'auto_load_libs': False},
-                     use_sim_procedures=True)
+    p = angr.Project(os.path.join(test_location, 'x86_64', 'datadep_test'),load_options={'auto_load_libs': False},use_sim_procedures=True)
```
Comment (twizmwazin) [inline tests/test_cdg.py:20]: Please avoid unnecessary reformatting

File: tests/test_constantpropagation.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
 
 def test_libc_x86():
 
-    p = angr.Project(os.path.join(test_location, "i386", "libc-2.27-3ubuntu1.so.6"), auto_load_libs=True)
+    p = angr.Project(os.path.join(test_location, "i386", "libc-2.27-3ubuntu1.so.6"), auto_load_libs=True)#disabling auto_load_libs increases the execution time.
```
Comment (twizmwazin) [inline tests/test_constantpropagation.py:12]: I think the comment would be better on the line above this. Also please have whitespace around the `#`.

File: tests/test_slicing.py [hunk 3]
```diff
@@ -14,9 +14,7 @@
 
 
 def test_find_exits():
-    slicing_test = angr.Project(os.path.join(test_location, 'x86_64', 'cfg_1'),
-                                use_sim_procedures=True,
-                                default_analysis_mode='symbolic')
+    slicing_test = angr.Project(os.path.join(test_location, 'x86_64', 'cfg_1'),use_sim_procedures=True, default_analysis_mode='symbolic', auto_load_libs=False)
```
Comment (twizmwazin) [inline tests/test_slicing.py:17]: Please keep these arguments on multiple lines and append rather than reformatting.

**Review by twizmwazin — CHANGES_REQUESTED** [review]
Please fix a few formatting issues, I left one comment per type of issue I noticed.

File: tests/test_cdg.py (hunk 1 above)
Comment (mohitrpatil) [reply tests/test_cdg.py:20]: I have fixed the reformatting issue. please let me know if something is missing.

File: tests/test_constantpropagation.py (hunk 2 above)
Comment (mohitrpatil) [reply tests/test_constantpropagation.py:12]: Changes done.

File: tests/test_slicing.py (hunk 3 above)
Comment (mohitrpatil) [reply tests/test_slicing.py:17]: Changes done. please review.

File: tests/test_accuracy.py [hunk 4]
```diff
@@ -16,7 +16,7 @@
 }
 
 def emulate(arch, binary, use_sim_procs, steps, hit_addrs, finished):
-    p = angr.Project(os.path.join(test_location, arch, binary), use_sim_procedures=use_sim_procs, rebase_granularity=0x1000000, load_debug_info=False)
+    p = angr.Project(os.path.join(test_location, arch, binary), use_sim_procedures=use_sim_procs, rebase_granularity=0x1000000, load_debug_info=False, auto_load_libs=True)# auto_load_libs can't be disabled as the test takes longer time to execute
```
Comment (twizmwazin) [inline tests/test_accuracy.py:19]: This line is way too long. Let's reformat it so each parameter to `Project` is on its own line. Also, `#` should always have spaces around it.

File: tests/test_accuracy.py [hunk 5]
```diff
@@ -59,7 +59,7 @@ def test_windows():
     yield emulate, 'i386', 'test_arrays.exe', True, 41, [], False # blocked on GetLastError or possibly dynamic loading
 
 def test_locale():
-    p = angr.Project(os.path.join(test_location, 'i386', 'isalnum'), use_sim_procedures=False)
+    p = angr.Project(os.path.join(test_location, 'i386', 'isalnum'), use_sim_procedures=False, auto_load_libs=True)# auto_load_libs can't be disabled as the test takes longer time to execute
```
Comment (twizmwazin) [inline tests/test_accuracy.py:62]: Same as above

File: tests/test_block_simplifier.py [hunk 6]
```diff
@@ -11,8 +11,7 @@
 
 
 def block_simplify(block):
-    p = angr.Project(os.path.join(os.path.dirname(os.path.abspath(__file__)),
-        '..', '..', 'binaries', 'tests', 'x86_64', 'fauxware'), auto_load_libs=False)
+    p = angr.Project(os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..', 'binaries', 'tests', 'x86_64', 'fauxware'), auto_load_libs=False)
```
Comment (twizmwazin) [inline tests/test_block_simplifier.py:14]: Unnecessary reformatting

File: tests/test_boyscout.py [hunk 7]
```diff
@@ -44,7 +44,7 @@ def main(file_path, arch, endianness):
                 'arch': 'ARM',
                 'offset': 0,
             }
-        }
+        }, auto_load_libs=False
```
Comment (twizmwazin) [inline tests/test_boyscout.py:47]: I think it would be more consistent to put this argument on a new line.

File: tests/test_cdg.py (hunk 1 above)
Comment (twizmwazin) [reply tests/test_cdg.py:20]: This isn't fixed. The original formatting for this function was fine, so simply reverting the changes to as they were is ideal.

File: tests/test_cfgemulated.py [hunk 8]
```diff
@@ -81,10 +81,7 @@ def get_function_name(addr):
             l.error("Edge (%s-0x%x, %s-0x%x) only exists in angr's CFG.", get_function_name(src), src, get_function_name(dst), dst)
 
 def perform_single(binary_path, cfg_path=None):
-    proj = angr.Project(binary_path,
-                        use_sim_procedures=True,
-                        default_analysis_mode='symbolic',
-                        load_options={'auto_load_libs': False})
+    proj = angr.Project(binary_path, use_sim_procedures=True, default_analysis_mode='symbolic',load_options={'auto_load_libs': False})
```
Comment (twizmwazin) [inline tests/test_cfgemulated.py:84]: unnecessary reformatting

File: tests/test_cfgemulated.py [hunk 9]
```diff
@@ -134,10 +131,7 @@ def test_additional_edges():
     # Test the `additional_edges` parameter for CFG generation
 
     binary_path = os.path.join(test_location, 'x86_64', 'switch')
-    proj = angr.Project(binary_path,
-                        use_sim_procedures=True,
-                        default_analysis_mode='symbolic',
-                        load_options={'auto_load_libs': False})
+    proj = angr.Project(binary_path,use_sim_procedures=True,default_analysis_mode='symbolic',load_options={'auto_load_libs': False})
```
Comment (twizmwazin) [inline tests/test_cfgemulated.py:134]: unnecessary reformatting

File: tests/test_cfgemulated.py [hunk 10]
```diff
@@ -156,10 +150,7 @@ def test_not_returning():
     # Make sure we are properly labeling functions that do not return in function manager
 
     binary_path = os.path.join(test_location, 'x86_64', 'not_returning')
-    proj = angr.Project(binary_path,
-                        use_sim_procedures=True,
-                        load_options={'auto_load_libs': False}
-                        )
+    proj = angr.Project(binary_path,use_sim_procedures=True,load_options={'auto_load_libs': False})
```
Comment (twizmwazin) [inline tests/test_cfgemulated.py:153]: unnecessary reformatting

File: tests/test_cfgemulated.py [hunk 11]
```diff
@@ -197,9 +188,7 @@ def test_cfg_6():
     # We need to add DO_CCALLS to resolve long jmp and support real mode
     o.modes['fastpath'] |= {o.DO_CCALLS}
     binary_path = test_location + "/i386/bios.bin.elf"
-    proj = angr.Project(binary_path,
-                        use_sim_procedures=True,
-                        page_size=1)
+    proj = angr.Project(binary_path, use_sim_procedures=True, page_size=1, auto_load_libs=False)
```
Comment (twizmwazin) [inline tests/test_cfgemulated.py:191]: unnecessary reformatting

File: tests/test_cfgemulated.py [hunk 12]
```diff
@@ -213,8 +202,8 @@ def test_fauxware():
 def disabled_loop_unrolling():
     binary_path = os.path.join(test_location, 'x86_64', 'cfg_loop_unrolling')
 
-    p = angr.Project(binary_path)
-    cfg = p.analyses.CFGEmulated(fail_fast=True)
+    p = angr.Project(binary_path, auto_load_libs=False)
+    cfg = p.analyses.CFGEmulated(fail_fast=False)
```
Comment (twizmwazin) [inline tests/test_cfgemulated.py:206]: Why was the fail_fast argument changed?

File: tests/test_constantpropagation.py (hunk 2 above)
Comment (twizmwazin) [reply tests/test_constantpropagation.py:12]: They do not appear to be

File: tests/test_cfgfast.py [hunk 13]
```diff
@@ -556,7 +556,7 @@ def test_serialization_cfgfast():
 
 def test_cfg_copy():
     path = os.path.join(test_location, "cgc", "CADET_00002")
-    proj = angr.Project(path)
+    proj = angr.Project(path, auto_load_libs= False)
```
Comment (twizmwazin) [inline tests/test_cfgfast.py:559]: Extra space

File: tests/test_ddg.py [hunk 14]
```diff
@@ -14,10 +14,7 @@
 test_location = os.path.join(os.path.dirname(os.path.realpath(__file__)), '..', '..', 'binaries', 'tests')
 
 def perform_one(binary_path):
-    proj = angr.Project(binary_path,
-                        load_options={'auto_load_libs': False},
-                        use_sim_procedures=True,
-                        default_analysis_mode='symbolic')
+    proj = angr.Project(binary_path, load_options={'auto_load_libs': False}, use_sim_procedures=True, default_analysis_mode='symbolic')
```
Comment (twizmwazin) [inline tests/test_ddg.py:17]: Unnecessary reformatting

File: tests/test_echo.py [hunk 15]
```diff
@@ -15,7 +15,7 @@
 }
 
 def run_echo_haha(arch):
-    p = angr.Project(os.path.join(test_location, arch, 'echo'), use_sim_procedures=False)
+    p = angr.Project(os.path.join(test_location, arch, 'echo'), use_sim_procedures=False)#auto_load_libs can't be disabled as the test fails
```
Comment (twizmwazin) [inline tests/test_echo.py:18]: Spaces around `#`

File: tests/test_jumptables.py [hunk 16]
```diff
@@ -290,8 +290,7 @@ def test_jumptable_occupied_as_data():
 
 def test_vtable_amd64_libc_ubuntu_2004():
 
-    p = angr.Project(os.path.join(test_location, "x86_64", "elf_with_static_libc_ubuntu_2004_stripped"),
-                     auto_load_libs=False)
+    p = angr.Project(os.path.join(test_location, "x86_64", "elf_with_static_libc_ubuntu_2004_stripped"),auto_load_libs=False)
```
Comment (twizmwazin) [inline tests/test_jumptables.py:293]: Unnecessary reformatting

File: tests/test_kb_plugins_dwarf.py [hunk 17]
```diff
@@ -6,10 +6,7 @@
 
 
 def test_kb_plugins_dwarf():
-    p = angr.Project(os.path.join(location, 'x86_64', 'state_merge_0'), load_options={
-        "load_debug_info":True,
-        "auto_load_libs": False,
-    } )
+    p = angr.Project(os.path.join(location, 'x86_64', 'state_merge_0'), load_options={"load_debug_info":True,"auto_load_libs": False,} )
```
Comment (twizmwazin) [inline tests/test_kb_plugins_dwarf.py:9]: Unnecessary reformatting

File: tests/test_mem_funcs.py [hunk 18]
```diff
@@ -8,21 +8,21 @@
 test_location = os.path.join(os.path.dirname(os.path.realpath(__file__)), '..', '..', 'binaries', 'tests')
 
 def test_memmove():
-    proj = angr.Project(os.path.join(test_location, 'x86_64', 'memmove'), load_options={'auto_load_libs': True}, exclude_sim_procedures_list=['memmove'])
+    proj = angr.Project(os.path.join(test_location, 'x86_64', 'memmove'), load_options={'auto_load_libs': True}, exclude_sim_procedures_list=['memmove'])#auto_load_libs can't be disabled as the testcase fails
```
Comment (twizmwazin) [inline tests/test_mem_funcs.py:11]: Each of the comments here should go on different lines

File: tests/test_multi_open_file.py [hunk 19]
```diff
@@ -9,7 +9,7 @@
 
 def test_multi_open_file():
     test_bin = os.path.join(test_location, "..", "..", "binaries", "tests", "x86_64", "test_multi_open_file")
-    b = angr.Project(test_bin)
+    b = angr.Project(test_bin) #auto_load_libs cannot be disabled as the test fails
```
Comment (twizmwazin) [inline tests/test_multi_open_file.py:12]: Spaces around `#`

File: tests/test_project_resolve_simproc.py [hunk 20]
```diff
@@ -11,7 +11,7 @@
 """
 
 def test_bina():
-    p = angr.Project(bina, exclude_sim_procedures_list=['rand', 'sleep'], load_options={"auto_load_libs":True})
+    p = angr.Project(bina, exclude_sim_procedures_list=['rand', 'sleep'], load_options={"auto_load_libs":True})#auto_load_libs can't be disabled as the testcase fails
```
Comment (twizmwazin) [inline tests/test_project_resolve_simproc.py:14]: Line length

File: tests/test_slicing.py (hunk 3 above)
Comment (twizmwazin) [reply tests/test_slicing.py:17]: Does not appear to be, also note that this applies to the other cases in this file

File: tests/test_stochastic.py [hunk 21]
```diff
@@ -15,8 +15,7 @@
 }
 
 def run_stochastic(binary, arch):
-    proj = angr.Project(os.path.join(location, arch, binary),
-                        auto_load_libs=False)
+    proj = angr.Project(os.path.join(location, arch, binary), auto_load_libs=False)
```
Comment (twizmwazin) [inline tests/test_stochastic.py:18]: Unnecessary reformatting

File: tests/test_str_funcs.py [hunk 22]
```diff
@@ -8,15 +8,15 @@
 test_location = os.path.join(os.path.dirname(os.path.realpath(__file__)), '..', '..', 'binaries', 'tests')
 
 def test_strncpy():
-    strncpy_amd64 = angr.Project(os.path.join(test_location, 'x86_64', 'strncpy'), load_options={'auto_load_libs': True}, exclude_sim_procedures_list=['strncpy'])
+    strncpy_amd64 = angr.Project(os.path.join(test_location, 'x86_64', 'strncpy'), load_options={'auto_load_libs': True}, exclude_sim_procedures_list=['strncpy'])#auto_load_libs can't be disabled as the test cases failed.
```
Comment (twizmwazin) [inline tests/test_str_funcs.py:11]: Line length for each change in this file

File: tests/test_strtol.py [hunk 23]
```diff
@@ -14,7 +14,7 @@ class TestStrtol(unittest.TestCase):
     @unittest.skipUnless(sys.platform.startswith("linux"), "linux-only")
     def test_strtol(self, threads=None):
         test_bin = os.path.join(bin_location, "tests", "x86_64", "strtol_test")
-        b = angr.Project(test_bin)
+        b = angr.Project(test_bin, auto_load_libs=True)#disabling auto_load_libs increases the execution time of the test case.
```
Comment (twizmwazin) [inline tests/test_strtol.py:17]: Spaces around `#`

File: tests/test_structurer.py [hunk 24]
```diff
@@ -178,8 +178,7 @@ def test_simple_loop():
 
 
 def test_recursive_structuring():
-    p = angr.Project(os.path.join(test_location, 'x86_64', 'cfg_loop_unrolling'),
-                     auto_load_libs=False, load_debug_info=True)
+    p = angr.Project(os.path.join(test_location, 'x86_64', 'cfg_loop_unrolling'),auto_load_libs=False, load_debug_info=True)
```
Comment (twizmwazin) [inline tests/test_structurer.py:181]: Unnecessary reformatting for each in this file

File: tests/test_syscall_override.py [hunk 25]
```diff
@@ -32,7 +32,7 @@
 }
 
 def run_fauxware_override(arch):
-    p = angr.Project(os.path.join(test_location, arch, "fauxware"), use_sim_procedures=False)
+    p = angr.Project(os.path.join(test_location, arch, "fauxware"), use_sim_procedures=False) #auto_load_libs can't be disabled as the test fails
```
Comment (twizmwazin) [inline tests/test_syscall_override.py:35]: Spaces around `#`

File: tests/test_unique.py [hunk 26]
```diff
@@ -17,8 +17,7 @@
 }
 
 def run_unique(binary, arch):
-    proj = angr.Project(os.path.join(location, arch, binary),
-                        auto_load_libs=False)
+    proj = angr.Project(os.path.join(location, arch, binary), auto_load_libs=False)
```
Comment (twizmwazin) [inline tests/test_unique.py:20]: Unnecessary reformatting

File: tests/test_veritesting.py [hunk 27]
```diff
@@ -22,10 +22,7 @@ def run_veritesting_a(arch):
 
     #logging.getLogger('angr.analyses.sse').setLevel(logging.DEBUG)
 
-    proj = angr.Project(os.path.join(location, arch, "veritesting_a"),
-                        load_options={'auto_load_libs': False},
-                        use_sim_procedures=True
-                        )
+    proj = angr.Project(os.path.join(location, arch, "veritesting_a"), load_options={'auto_load_libs': False}, use_sim_procedures=True)
```
Comment (twizmwazin) [inline tests/test_veritesting.py:25]: Unnecessary reformatting x2

File: tests/test_vfg.py [hunk 28]
```diff
@@ -17,9 +17,7 @@
 }
 
 def run_vfg_buffer_overflow(arch):
-    proj = angr.Project(os.path.join(test_location, arch, "basic_buffer_overflows"),
-                 use_sim_procedures=True,
-                 default_analysis_mode='symbolic')
+    proj = angr.Project(os.path.join(test_location, arch, "basic_buffer_overflows"), use_sim_procedures=True, default_analysis_mode='symbolic', auto_load_libs=False)
```
Comment (twizmwazin) [inline tests/test_vfg.py:20]: Unnecessary reformatting x2

**Review by twizmwazin — CHANGES_REQUESTED** [review]
Lots of little code formatting things

File: tests/test_slicing.py [hunk 29]
```diff
@@ -47,9 +48,7 @@ def test_find_exits():
 
 
 def test_control_flow_slicing():
-    slicing_test = angr.Project(os.path.join(test_location, 'x86_64', 'cfg_1'),
-                                use_sim_procedures=True,
-                                default_analysis_mode='symbolic')
+    slicing_test = angr.Project(os.path.join(test_location, 'x86_64', 'cfg_1'),use_sim_procedures=True,default_analysis_mode='symbolic', auto_load_libs=False)
```
Comment (twizmwazin) [inline tests/test_slicing.py:51]: Reformatting

File: tests/test_slicing.py [hunk 30]
```diff
@@ -69,9 +68,7 @@ def broken_backward_slice():
 
     # TODO: Fix this test case
 
-    slicing_test = angr.Project(os.path.join(test_location, "x86_64", "cfg_1"),
-                                use_sim_procedures=True,
-                                default_analysis_mode='symbolic')
+    slicing_test = angr.Project(os.path.join(test_location, "x86_64", "cfg_1"),use_sim_procedures=True,default_analysis_mode='symbolic', auto_load_libs=False)
```
Comment (twizmwazin) [inline tests/test_slicing.py:71]: Reformatting

File: tests/test_slicing.py [hunk 31]
```diff
@@ -103,8 +100,7 @@ def broken_backward_slice():
 
 
 def test_last_branching_statement():
-    slicing_test = angr.Project(os.path.join(test_location, 'armel', 'fauxware'),
-                                use_sim_procedures=True)
+    slicing_test = angr.Project(os.path.join(test_location, 'armel', 'fauxware'),use_sim_procedures=True, auto_load_libs=False)
```
Comment (twizmwazin) [inline tests/test_slicing.py:103]: Reformatting

File: tests/test_vfg.py [hunk 32]
```diff
@@ -18,8 +18,9 @@
 
 def run_vfg_buffer_overflow(arch):
     proj = angr.Project(os.path.join(test_location, arch, "basic_buffer_overflows"),
-                 use_sim_procedures=True,
-                 default_analysis_mode='symbolic')
+                use_sim_procedures=True,
+                default_analysis_mode='symbolic',
```
Comment (twizmwazin) [inline tests/test_vfg.py:22]: Reformatting, short 1 space

File: tests/test_vfg.py [hunk 33]
```diff
@@ -124,10 +125,9 @@ def run_vfg_0(arch):
 }
 
 def run_vfg_1(arch):
-    proj = angr.Project(
-        os.path.join(test_location, arch, "fauxware"),
-        use_sim_procedures=True,
-    )
+    proj = angr.Project(os.path.join(test_location, arch, "fauxware"),
+                        use_sim_procedures=True, 
+                        auto_load_libs=False)
```
Comment (twizmwazin) [inline tests/test_vfg.py:130]: Reformatting

**Review by twizmwazin — CHANGES_REQUESTED** [review]
Almost there!

**Review by twizmwazin — APPROVED** [review]

**Comment by ltfish** [comment]
Merged. Thanks!

**Pull request merged**